tests/hypothesis/to_unit_test.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+import pytest
+from hypothesis import given
+from hypothesis import strategies as st
+
+import scipp as sc
+
+
+@given(st.floats(min_value=1e-300, max_value=1e300))
+def test_to_unit_value_1(x):
+    unit = sc.Unit(f'{x:.12E}')
+    var = sc.scalar(1.23 * x, unit='')
+    result = sc.to_unit(var, unit)
+    assert result.unit == unit
+    assert pytest.approx(result.value, 1e-14) == 1.23
```

Comment:
Why 1e-14? Given that `unit` is only encoded to 13 digits, I would not expect the test to work with this tolerance. But apparently it does?

Reply:
🤦 while `pytest.approx` has a default of `abs=None`, it uses `1e-12` as default absolute error bound